frontend/src/components/DropDownGroup.jsx

```diff
@@ -0,0 +1,205 @@
+import { Menu } from "antd";
+import { useState, useEffect } from "react";
+import { NavLink } from "react-router-dom";
+import { useContext } from "react";
+import { AuthContext } from "../components/AuthContext";
+import Logout from "../components/Logout";
+
+const DropDownGroup = () => {
+  const [current, setCurrent] = useState("");
+  const { user, loading } = useContext(AuthContext);
+  const [links, setLinks] = useState();
+
+  const onClick = (e) => {
+    setCurrent(e.key);
+  };
+
+  const userCheckHours = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Kaikki tunnit"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckCumulativeHours = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Kaikki tunnit kumulatiivinen"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckSalary = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Palkka"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckBillability = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Laskutettavat tunnit"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+
+  const userCheckAdmin = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedAdmin = parsedClaims["admin"];
+        return parsedAdmin;
+      } catch (error) {
+        return null;
+      }
+    }
+    //
+  };
```

Comment:
It would really reduce the number of code lines if you came up with a generic function that solves all these cases. Can you check these and figure out what changes (apart from variable names)? How do you create a single function that could be used to check all different permissions?